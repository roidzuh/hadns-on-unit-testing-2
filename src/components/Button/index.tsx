import React from 'react';
import { Props } from './types.type';

const Button = (props: Props) => {
  const { text, onclick, isDisable } = props;
  return (
    <button disabled={isDisable} onClick={onclick}>
      {text}
    </button>
  );
};

export default Button;
