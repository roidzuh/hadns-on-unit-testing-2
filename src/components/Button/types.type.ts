export type Props = {
  text: string;
  onclick: () => void;
  isDisable?: boolean;
};
