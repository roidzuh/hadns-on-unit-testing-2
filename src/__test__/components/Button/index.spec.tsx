import React from 'react';
import { render } from '@testing-library/react';
import Button from '@/components/Button';

describe('button component', () => {
  it('should handle render button not disable', () => {
    const { container } = render(
      <Button text={'button'} onclick={() => alert('hello world')} />
    );
    expect(container).toMatchSnapshot();
  });

  it('should handle render button disable', () => {
    const { container } = render(
      <Button isDisable text={'button'} onclick={() => alert('hello world')} />
    );
    expect(container).toMatchSnapshot();
  });
});
