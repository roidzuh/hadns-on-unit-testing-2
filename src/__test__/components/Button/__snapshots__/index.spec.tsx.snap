// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component should handle render button disable 1`] = `
<div>
  <button
    disabled=""
  >
    button
  </button>
</div>
`;

exports[`button component should handle render button not disable 1`] = `
<div>
  <button>
    button
  </button>
</div>
`;
